src/main/java/space/morphanone/webizen/events/BasicRequestScriptEvent.java

```diff
@@ -178,7 +180,19 @@ public ObjectTag getContext(String name) {
             return new ElementTag(httpExchange.getRemoteAddress().toString());
         }
         else if (name.equals("query")) {
-            return new ElementTag(httpExchange.getRequestURI().getQuery());
+            String query = httpExchange.getRequestURI().getQuery();
+            return new ElementTag(query != null ? query : "");
+        }
+        else if (name.equals("query_map")) {
+            MapTag mappedValues = new MapTag();
+            String query = httpExchange.getRequestURI().getQuery();
+            if (query != null) {
+                for (String value : CoreUtilities.split(query, '&')) {
+                    List<String> split = CoreUtilities.split(value, '=');
+                    mappedValues.map.put(new StringHolder(split.get(0)), new ElementTag(split.get(1)));
```

Comment:
This needs to either be replaced with an appropriate query map usage of the underlying API (Instead of parsing the string), or have querystring unescaping applied